rialto_airflow/harvest/pubmed.py

```diff
@@ -213,29 +213,22 @@ def _pubmed_search_api(query) -> list:
 
     full_url = f"{BASE_URL}{SEARCH_PATH}&api_key={pubmed_key()}"
     logging.debug(f"searching pubmed with {params}")
-    try:
-        response = requests.get(full_url, params=params, headers=HEADERS)
-        response.raise_for_status()
-
-        results = response.json()
-
-        if "error" in results:
-            logging.error(f"Error in results found for {query}: {results['error']}")
-            return []
-
-        if results.get("esearchresult", {}).get("count") is None:
-            logging.debug(f"No esearchresult or count found for {query}")
-            return []
 
-        if int(results["esearchresult"]["count"]) == 0:
-            logging.debug(f"No results found for {query}")
-            return []
-
-        return results["esearchresult"]["idlist"]  # return a list of pmids
+    response = requests.get(full_url, params=params, headers=HEADERS)
+    response.raise_for_status()
+    results = response.json()
 
-    except requests.exceptions.RequestException as e:  # Catch all requests exceptions
-        logging.error(f"Error searching pubmed query {params}: {e}")
+    if "error" in results:
+        logging.error(f"Error in results found for {query}: {results['error']}")
         return []
+    elif results.get("esearchresult", {}).get("count") is None:
+        logging.debug(f"No esearchresult or count found for {query}")
+        return []
+    elif int(results["esearchresult"]["count"]) == 0:
+        logging.debug(f"No results found for {query}")
+        return []
+    else:
+        return results["esearchresult"]["idlist"]  # return a list of pmids
```

Comment:
HTTP level exceptions will now propagate, and halt harvesting. Maybe we'll have to revisit this, but it would be good to know when something is going wrong instead of things failing and getting logged.